### ISSUE TITLE
treesheets segfaults when built from source, but not when installed from cache

### ISSUE THREAD

**Issue by AveryLychee** [NixOS/nixpkgs#44259]
## Issue description

the treesheets application segfaults when built from source, but not when installed from the cache.

I discovered this as I was trying to update it to the latest version - which should be a relatively simple change, but even the version that is available on the nixos cache doesn't work when compiled from source.

I have contributed to treesheets, and recently built it on both Manjaro and Sabayon linux, so it's not an issue upstream.

### Steps to reproduce

1. `nix-env -iA nixos.treesheets`
2. run `treesheets`, it should open the program
3. `nix-env -f '/path/to/nixpkgs/source' --option binary-caches "" -iA treesheets`
4. run `treesheets`, it should output "treesheets terminated by signal SIGSEGV (Address boundary error)" (or perhaps a less descriptive error in a different shell)

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.54, NixOS, 18.03.132847.aec217852f2 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by dotlambda** [comment]
Cannot reproduce. Please try again with sandboxing enabled.

**Comment by AveryGlitch** [comment]
(Sorry for the daft question, but because nixos has already built a version of treesheets with the same hash, it's just installing that - how do I force it to rebuild from source?)

**Comment by dotlambda** [comment]
There are many ways. An easy one would be to modify the expression just a tiny bit without really modifying the package.

**Comment by AveryGlitch** [comment]
treesheets built from source still segfaults with sandboxing on (regardless of whether I use the version in this repository or the latest version)

**Comment by danbst** [comment]
Here is nice way to show reproduction steps:
```
$ nix-build https://github.com/nixos/nixpkgs/archive/aec217852f2.tar.gz -A treesheets --check
$ ./result/bin/treesheets
```
Except that you should specify proper nixpkgs checkout revision. The `--check` will force rebuild locally.

That said, I've reproduced the issue with above commands even without sandboxing.
```
'g_io_module_load': /nix/store/hy4j0i8kqhydmwbwyx2wq5rfjvad1dvf-glib-networking-2.56.0/lib/gio/modules/libgiognutls.so: undefined symbol: g_io_module_load
Failed to load module: /nix/store/hy4j0i8kqhydmwbwyx2wq5rfjvad1dvf-glib-networking-2.56.0/lib/gio/modules/libgiognutls.so
'g_io_module_load': /nix/store/hy4j0i8kqhydmwbwyx2wq5rfjvad1dvf-glib-networking-2.56.0/lib/gio/modules/libgiolibproxy.so: undefined symbol: g_io_module_load
Failed to load module: /nix/store/hy4j0i8kqhydmwbwyx2wq5rfjvad1dvf-glib-networking-2.56.0/lib/gio/modules/libgiolibproxy.so
'g_io_module_load': /nix/store/hy4j0i8kqhydmwbwyx2wq5rfjvad1dvf-glib-networking-2.56.0/lib/gio/modules/libgiognomeproxy.so: undefined symbol: g_io_module_load
Failed to load module: /nix/store/hy4j0i8kqhydmwbwyx2wq5rfjvad1dvf-glib-networking-2.56.0/lib/gio/modules/libgiognomeproxy.so
fish: “./result/bin/treesheets” terminated by signal SIGSEGV (Address boundary error)
```

**Comment by AveryGlitch** [comment]
It gets stranger. So I downloaded the latest version of the source code from git, then put a default.nix file in the directory:

```nix
with import <nixpkgs> {};
stdenv.mkDerivation rec {
  name    = "treesheets-${version}";
  version = "2018-07-20";

  src = fetchFromGitHub {
    owner  = "aardappel";
    repo   = "treesheets";
    rev    = "52aa49ef12430ecb33535bcaabcdd2119801e734";
    sha256 = "02q3fyh6a3c8cfkp2j13qpca486nj0f080xrx4hdzczrzmhqhhmm";
  };

  buildInputs = [ wxGTK31 makeWrapper ];

  preConfigure = "cd src";

  postInstall = ''
    mkdir "$out/share" -p
    cp -av ../TS "$out/share/libexec"

    mkdir "$out/bin" -p
    makeWrapper "$out/share/libexec/treesheets" "$out/bin/treesheets"

    mkdir "$out/share/doc" -p

    for f in readme.html docs examples
    do
      mv -v "$out/share/libexec/$f" "$out/share/doc"
      ln -sv "$out/share/doc/$f" "$out/share/libexec/$f"
    done

    mkdir "$out/share/applications" -p
    mv -v "$out/share/libexec/treesheets.desktop" "$out/share/applications"
    substituteInPlace "$out/share/applications/treesheets.desktop" \
      --replace "Icon=images/treesheets.svg" "Icon=$out/share/libexec/images/treesheets.svg"
  '';

  meta = with stdenv.lib; {
    description = "Free Form Data Organizer";

    longDescription = ''
      The ultimate replacement for spreadsheets, mind mappers, outliners,
      PIMs, text editors and small databases.

      Suitable for any kind of data organization, such as Todo lists,
      calendars, project management, brainstorming, organizing ideas,
      planning, requirements gathering, presentation of information, etc.
    '';

    homepage    = http://strlen.com/treesheets/;
    maintainers = with maintainers; [ obadz ];
    platforms   = platforms.linux;
    license     = licenses.zlib;
  };
}
```

then, if I type `nix-build` and then `./result/bin/treesheets`, the program runs as expected.

But if I take the very same file and put it into the nixpkgs repository (which was up to date with master as of 1 week ago), the resulting executable segfaults.

(side note: in order to put this file into the nixpkgs repository, you need to set `wxGTK` to `wxGTK31`, which requires an edit to `all-packages.nix`)

**Comment by AveryLychee** [comment]
Fixed by #47259
